### PR TITLE
Editor revision 2021 03 23

### PR DESCRIPTION
**Pull request by sthagen**
A Rendered view on the resulting document is available by visiting https://github.com/sthagen/csaf/blob/editor-revision-2021-03-23/csaf_2.0/prose/csaf-v2-editor-draft.md

@santosomar: Can we merge the PR (as requested) into the `oasis-tcs:editor-revision-2021-03-23` branch before the meeting for the reading pleasure of the members and then create a PR into `oasis-tcs:master` branch for discussion?
Please kindly advise.

### CODE REVIEW

**Comment by tschmidtb51** [comment]
Thank you for your work.

**Comment by sthagen** [comment]
> Thank you for your work.
You are welcome. The final parts  in section 3 - and revised sections 4+ will follow before midnight CET - two weeks late and took twice as long ... but OK in physics there is some magic associated with a factor of 2 ...

**Comment by sthagen** [comment]
@chet-ensign: I added you to reviewers to request kind feedback on some "technical" changes I made. Some of these are:
* I replaced the pesky `non-normative` term with `informative` (as counterpart to `normative`)
* We now have a quite packed conformance section matching typical roles in the existing complex security advisory sharing communities

In addition to the second point: When would it make sense to kindly request review of these from the Technical Advisory Board (TAB) of OASIS such that we can work on such feedback in parallel and do not risk additional cycles after public comment periods?

Thanks a lot in advance,
Stefan

**Comment by OASIS-OP-Admin** [comment]
@sthagen - Thanks. Changing 'non-normative' to 'informative' is fine. I agree that it sounds more natural next to 'Normative'

The conformance section really is extensive. While the TAB has not been reviewing work for some time, if you would like to invite comments, I see no reason not to go with this version. Jacques in particular might be interested to see what you have done.

**Comment by sthagen** [comment]
I will do so. Thank you so much for giving actionable advice so fast, @chet-ensign

**Review by chet-ensign — APPROVED** [review]

**Review by santosomar — APPROVED** [review]
Thank you @sthagen for changing this and @chet-ensign for your continuous support!

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 1]
```diff
@@ -608,44 +608,161 @@ Valid `enum` values are:
...
+This package URL (PURL) attribute refers to a method for reliably identifying and locating software packages external to this specification.
+
+##### 3.1.3.3.4 Full Product Name Type - Product Identification Helper - Serial Numbers
+
+The list of serial numbers representation (`serial_numbers`) of value type `array` with 1 or more items contains a list of parts, or a full serial numbers.
+
+A list of serial numbers SHOULD only be used if a certain range of serial numbers with a software version is affected, or the serial numbers change during update.
+
+Any given serial number of value type `string` with at least 1 character represents a part, or a full serial number of the component to identify.
+
```
Comment (tschmidtb51) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:731]: We should add a remark here: 
If a part of a serial number of the component to identify is given it SHOULD begin with the first character of the serial number and stop at any point. Characters which should not be matched MUST be replaced by either `?` (for a single character) or `*` (for 0 or multiple characters). Two `*` MUST NOT follow each other.

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 2]
```diff
@@ -608,44 +608,161 @@ Valid `enum` values are:
...
+    }
+
+
+##### 3.1.3.3.5 Full Product Name Type - Product Identification Helper - Generic URIs
+
+List of generic URIs representation (`x_generic_uris`) of value type `array` with at least 1 item contains a list of identifiers which are either vendor-specific or derived from a standard not yet supported.
+
+Any such Generic URI item of value type `object` provides the two mandatory properties `namespace` and URI (`uri`).
+
+The Common Platform Enumeration (CPE) attribute refers to a method for naming platforms external to this specification.
```
Comment (tschmidtb51) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:746]: This line should not be here.

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 3]
```diff
@@ -608,44 +608,161 @@ Valid `enum` values are:
...
     ^(cpe:2\\.3:[aho\\*\\-](:(((\\?*|\\*?)([a-zA-Z0-9\\-\\._]|(\\\\[\\\\\\*\\?!\"#\\$%&'\\(\\)\\+,/:;<=>@\\[\\]\\^`\\{\\|\\}~]))+(\\?*|\\*?))|[\\*\\-])){5}(:(([a-zA-Z]{2,3}(-([a-zA-Z]{2}|[0-9]{3}))?)|[\\*\\-]))(:(((\\?*|\\*?)([a-zA-Z0-9\\-\\._]|(\\\\[\\\\\\*\\?!\"#\\$%&'\\(\\)\\+,/:;<=>@\\[\\]\\^`\\{\\|\\}~]))+(\\?*|\\*?))|[\\*\\-])){4})|([c][pP][eE]:/[AHOaho]?(:[A-Za-z0-9\\._\\-~%]*){0,6})$
 
 The Common Platform Enumeration (CPE) attribute refers to a method for naming platforms external to this specification.
 
-#### 3.1.3.2 Full Product Name Type - Name
+##### 3.1.3.3.2 Full Product Name Type - Product Identification Helper - Hashes
 
-Textual description of the product (`name`) has value type `string` with 1 or more characters.
-The value should be the product's full canonical name, including version number and other attributes, as it would be used in a human-friendly document.
+List of hashes representation (`hashes`) of value type `array` holding at least one item contains a list of cryptographic hashes usable to identify files.
```
Comment (tschmidtb51) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:680]: I would remove the word "representation" here.

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 4]
```diff
@@ -608,44 +608,161 @@ Valid `enum` values are:
...
+
+The package URL (PURL) representation (`purl`) is a `string` of 4 or more characters with `pattern` (regular expression):
+
+    ^pkg:
+
+This package URL (PURL) attribute refers to a method for reliably identifying and locating software packages external to this specification.
+
+##### 3.1.3.3.4 Full Product Name Type - Product Identification Helper - Serial Numbers
+
+The list of serial numbers representation (`serial_numbers`) of value type `array` with 1 or more items contains a list of parts, or a full serial numbers.
```
Comment (tschmidtb51) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:726]: I would remove the word "representation" here.

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 5]
```diff
@@ -608,44 +608,161 @@ Valid `enum` values are:
...
+        //...  
+      "items": {
+        //...  
+      }
+    }
+
+
+##### 3.1.3.3.5 Full Product Name Type - Product Identification Helper - Generic URIs
+
+List of generic URIs representation (`x_generic_uris`) of value type `array` with at least 1 item contains a list of identifiers which are either vendor-specific or derived from a standard not yet supported.
```
Comment (tschmidtb51) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:742]: I would remove the word "representation" here.

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 6]
```diff
@@ -608,44 +608,161 @@ Valid `enum` values are:
...
+
+Any such Generic URI item of value type `object` provides the two mandatory properties `namespace` and URI (`uri`).
```
Comment (tschmidtb51) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:744]: I would replace `namespace` with Namespace (`namespace`) or remove the word URI (and the brackets).

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 7]
```diff
@@ -1779,95 +1896,104 @@ CSAF documents are based on JSON, thus the security considerations of [RFC8259]
...
-
-While it may not be immediately obvious how your specification might make systems vulnerable to attack, most specifications, because they involve communications between systems, message formats, or system settings, open potential channels for exploit. For example, IETF [[RFC3552](#rfc3552)] lists “eavesdropping, replay, message insertion, deletion, modification, and man-in-the-middle” as well as potential denial of service attacks as threats that must be considered and, if appropriate, addressed in IETF RFCs.
-
-In addition to considering and describing foreseeable risks, this section should include guidance on how implementers and adopters can protect against these risks.
-
-We encourage editors and TC members concerned with this subject to read _Guidelines for Writing RFC Text on Security Considerations_, IETF [[RFC3552](#rfc3552)], for more information.
+In the only subsection of this section, the conformance targets and clauses are listed.
+The clauses matching the targets one to one, are listed in separate sub-subsections of the targets listing subsection.
```
Comment (tschmidtb51) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:1910]: I would add a comma here to support the reader: 
> The clauses, matching...

However, I'm not sure whether a native speaker would do that.

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 8]
```diff
@@ -1779,95 +1896,104 @@ CSAF documents are based on JSON, thus the security considerations of [RFC8259]
...
 * **CSAF management system**: A program that is able to manage CSAF documents and is able to display their details as required by CSAF viewer.
 * **CSAF asset matching system**: A program that connects to or is asset database and is able to manage CSAF documents as required by CSAF management system as well as matching them to assets of the asset database.
 
-## 5.2 Conformance Clause 1: CSAF document
+### 5.1.1 Conformance Clause 1: CSAF document
+
+A text file satisfies the "CSAF document" conformance profile if the text file:
 
-A text file satisfies the "CSAF document" conformance profile if:
+* conforms to the syntax and semantics defined in section 3.
```
Comment (tschmidtb51) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:1950]: We should add an additional point in regards to the file name.

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 9]
```diff
@@ -1779,95 +1896,104 @@ CSAF documents are based on JSON, thus the security considerations of [RFC8259]
...
-* It does not emit any objects, properties, or values which, according to section 3, are intended to be produced only by converters.
+### 5.1.4 Conformance Clause 4: Converter
 
-## 5.5 Conformance Clause 4: Converter
+A converter satisfies the “Converter” conformance profile if the converter:
 
-A converter satisfies the “Converter” conformance profile if:
+* satisfies the "CSAF producer" conformance profile.
+* additionally satisfies those normative requirements in section 3 that are designated as applying to converters.
+* does not emit any objects, properties, or values which, according to section 3, are intended to be produced only by direct producers.
```
Comment (tschmidtb51) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:1972]: Such normative requirements do not exist yet in section 3 (if I remember correctly). Do we want to introduce them?

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 10]
```diff
@@ -1779,95 +1896,104 @@ CSAF documents are based on JSON, thus the security considerations of [RFC8259]
...
-* It satisfies those normative requirements in section 3 that are designated as applying to CSAF producers.
+### 5.1.3 Conformance Clause 3: Direct producer
 
-## 5.4 Conformance Clause 3: Direct producer
+An analysis tool satisfies the "Direct producer" conformance profile if the analysis tool:
 
-An analysis tool satisfies the "Direct producer" conformance profile if:
+* satisfies the "CSAF producer" conformance profile.
+* additionally satisfies those normative requirements in section 3 that are designated as applying to "direct producers" or to "analysis tools".
+* does not emit any objects, properties, or values which, according to section 3, are intended to be produced only by converters.
```
Comment (tschmidtb51) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:1964]: Such normative requirements do not exist yet in section 3 (if I remember correctly). Do we want to introduce them?

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 11]
```diff
@@ -608,44 +608,161 @@ Valid `enum` values are:
...
+#### 3.1.3.3 Full Product Name Type - Product Identification Helper
+
+Helper to identify the product (`product_identification_helper`) of value type `object` provides in its properties at least one method which aids in identifying the product in an asset database.
+Of the given five properties `cpe`, `hashes`, `purl`, `serial_numbers`, and `x_generic_uris` one is mandatory.
+
+    "product_identification_helper": {
+      "title": "Helper to identify the product",
+      "description": "Provides at least one method which aids in identifying the product in an asset database.",
+      "type": "object",
+      "minProperties": 1,
```
Comment (tschmidtb51) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:651]: should be `// ...`

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 12]
```diff
@@ -1927,82 +2054,94 @@ A CSAF content management system satisfies the "CSAF content management system"
...
+      * `/document/tracking/version` with the value of `number` the latest `/document/tracking/revision_history[]` element
+      * `/document/publisher` and children
+
+### 5.1.7 Conformance Clause 7: CSAF post-processor
+
+A CSAF post-processor satisfies the "CSAF post-processor" conformance profile if the post-processor:
+
+* satisfies the "CSAF consumer" conformance profile.
+* satisfies the "CSAF producer" conformance profile.
+* additionally satisfies those normative requirements in section 3 that are designated as applying to post-processors.
```
Comment (tschmidtb51) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:2080]: Such normative requirements do not exist yet in section 3 (if I remember correctly). Do we want to introduce them?

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 13]
```diff
@@ -1927,82 +2054,94 @@ A CSAF content management system satisfies the "CSAF content management system"
...
+A program satisfies the "CSAF modifier" conformance profile if the program fulfills the two following groups of requirements:
 
-A CSAF post-processor satisfies the "CSAF post-processor" conformance profile if:
+The program:
 
-* It satisfies the "CSAF consumer" conformance profile.
-* It satisfies the "CSAF producer" conformance profile.
-* It additionally satisfies those normative requirements in section 3 that are designated as applying to post-processors.
+* satisfies the "CSAF post-processor" conformance profile.
+* adds, deletes or modifies at least one property or object or value of a property.
```
Comment (tschmidtb51) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:2089]: Shouldn't we mention here arrays as well?

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 14]
```diff
@@ -1927,82 +2054,94 @@ A CSAF content management system satisfies the "CSAF content management system"
...
+* provides the `/document/lang` property with a value matching the language of the translation.
+* provides the `/document/source_lang` to contain the language of the original document (and should only be set by CSAF tranlators).
+* includes a reference to the original advisory as first element of the array `/document/references[]`.
+* may contain translations for elements in arrays of `references_t` after the first element. However, it must keep the original urls as references at the end.
 
-## 5.11 Conformance Clause 10: CSAF consumer
+### 5.1.10 Conformance Clause 10: CSAF consumer
 
-A consumer satisfies the "CSAF consumer" conformance profile if:
+A consumer satisfies the "CSAF consumer" conformance profile if the consumer:
```
Comment (tschmidtb51) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:2119]: I would replace consumer with program:
> A program satisfies the "CSAF consumer" conformance profile if the program:

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 15]
```diff
@@ -1927,82 +2054,94 @@ A CSAF content management system satisfies the "CSAF content management system"
...
+* satisfies the normative requirements given below.
 
-## 5.13 Conformance Clause 12: CSAF management system
 
-A CSAF management system satisfies the "CSAF management system" conformance profile if:
+For each CVSS-Score in `/vulnerabilities[]/scores[]` the viewer:
 
-* It satisfies the "CSAF viewer" conformance profile.
-* It provides at least the following management functions:
+* preferably shows the `vector` if there is an inconsistency between the `vector' and any other sibling attribute.
```
Comment (tschmidtb51) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:2136]: run-away `
> * preferably shows the `vector` if there is an inconsistency between the `vector` and any other sibling attribute.

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 16]
```diff
@@ -2014,41 +2153,28 @@ A CSAF management system satisfies the "CSAF management system" conformance prof
...
+A CSAF asset matching system satisfies the "CSAF asset matching system" conformance profile if the asset matching system:
+
+* satisfies the "CSAF management system" conformance profile.
+* is an asset database or connects to one.
+* matches the CSAF documents within the system to the respective assets. This might be done with a probability which gives the end user the chance to broaden or narrow the results.
+* provides for each product of the asset database a list of matched advisories.
+* provides for each asset of the asset database a list of matched advisories.
+* provides for each CSAF document a list of matched product of the asset database.
+* provides for each CSAF document a list of matched asset of the asset database.
+* provides for each vulnerability within a CSAF document the option to mark matched asset of the asset database as "not remediated", "remediation in progress", "remediation done". A switch to mark all in the same status may be implemented.
```
Comment (tschmidtb51) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:2170]: I think it should be "at the same time".

**Review by tschmidtb51 — COMMENTED** [review]
@sthagen: Thank you for your great work! I had a few remarks...

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 12 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:2080]: I guess the best would be to refer to section 5 as we have done in the other cases.

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 9 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:1972]: I guess the best would be to refer to section 5 as we have done in the other cases.

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 10 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:1964]: I guess the best would be to refer to section 5 as we have done in the other cases.

**Review by mprpic — COMMENTED** [review]
@sthagen Couple more fixes below; I guess they should be applied to your branch and not on top of master.

**Comment by sthagen** [comment]
> @sthagen Couple more fixes ...; I guess they should be applied to your branch and not on top of master.
Yes, I will do so. Thanks a lot for reviewing @mprpic

File: csaf_2.0/json_schema/csaf_json_schema.json [hunk 17]
```diff
@@ -232,36 +232,36 @@
...
-              "title": "List of generic uris",
-              "description": "Contains a list of identifiers which are either vendor-specific or derivated from a standard not yet supported.",
+              "title": "List of generic URIs",
+              "description": "Contains a list of identifiers which are either vendor-specific or derived from a standard not yet supported.",
               "type": "array",
               "minItems": 1,
               "items": {
-                "title": "Generic uri",
-                "description": "Provides a generic extention point for any identifier which is either vendor-specific or derivated from a standard not yet supported.",
+                "title": "Generic UR",
```
[the remark this answers is not in the document]
Comment (tschmidtb51) [reply csaf_2.0/json_schema/csaf_json_schema.json:254]: Resolved with 7512432.

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 18]
```diff
@@ -137,14 +125,28 @@ For purposes of this document, the following terms and definitions apply:
 
 **converter**: CSAF producer that transforms the output of an analysis tool from its native output format into the CSAF format
 
+**CSAF asset matching system**: program that connects to or is asset database and is able to manage CSAF documents as required by CSAF management system as well as matching them to assets of the asset database.
```
[the remark this answers is not in the document]
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:128]: Resolved with 7512432.

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 19]
```diff
@@ -243,8 +245,10 @@ For purposes of this document, the following terms and definitions apply:
 
 **web analysis tool**: analysis tool that models and analyzes the interaction between a web client and a server.
```
[the remark this answers is not in the document]
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:246]: Resolved with 7512432.

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 20]
```diff
@@ -608,44 +608,161 @@ Valid `enum` values are:
...
 
 The Common Platform Enumeration (CPE) attribute refers to a method for naming platforms external to this specification.
 
-#### 3.1.3.2 Full Product Name Type - Name
+##### 3.1.3.3.2 Full Product Name Type - Product Identification Helper - Hashes
 
-Textual description of the product (`name`) has value type `string` with 1 or more characters.
-The value should be the product's full canonical name, including version number and other attributes, as it would be used in a human-friendly document.
+List of hashes representation (`hashes`) of value type `array` holding at least one item contains a list of cryptographic hashes usable to identify files.
+
```
[the remark this answers is not in the document]
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:681]: Resolved with 7512432.

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 21]
```diff
@@ -608,44 +608,161 @@ Valid `enum` values are:
...
+    WINWORD.EXE
+    msotadddin.dll
+    sudoers.so
+
+The Value of the cryptographic hash representation (`value`) of value type `string` of 64 or more characters with `pattern` (regular expression):
+
+    ^[0-9a-fA-F]{64,}$
+
+The Value of the cryptographic hash attribute ontains the cryptographic hash value.
```
[the remark this answers is not in the document]
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:708]: Resolved with 7512432.

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 15 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:2136]: Resolved with 7512432.

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 16 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:2170]: Resolved with 7512432.

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 14 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:2119]: Resolved with 7512432.

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 1 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:731]: Resolved with 7512432.

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 2 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:746]: Resolved with 7512432.

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 4 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:726]: Resolved with 7512432.

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 5 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:742]: Resolved with 7512432.

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 6 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:744]: Resolved with 7512432.

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 11 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:651]: Resolved with 7512432.

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 3 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:680]: Resolved with #200.

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 7 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:1910]: Was copied to #205 to have the discussion in a single place.

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 8 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:1950]: Was copied to #205 to have the discussion in a single place.

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 13 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:2089]: Resolved with #200.Resolved with #200.

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 22]
```diff
@@ -1927,82 +2054,94 @@ A CSAF content management system satisfies the "CSAF content management system"
...
-* The translated document must include a reference to the original advisory as first element of the array `/document/references[]`.
-* It may insert translations for elements in arrays of `references_t` after the first element. However, it must keep the original urls as references at the end.
+* has not the same `/document/tracking/id` as the original document. The translated document can use a completely new `/document/tracking/id` or compute one by using the original `/document/tracking/id` as a prefix and add an ID from the naming scheme of the issuer of the translated version. It should not use the original `/document/tracking/id` as a suffix. If an issuer uses an CSAF translator to publish his advisories in multiple languages he may use the combination of original `/document/tracking/id` and translated `/document/lang` as a `/document/tracking/id` for the translated document.
+* provides the `/document/lang` property with a value matching the language of the translation.
+* provides the `/document/source_lang` to contain the language of the original document (and should only be set by CSAF tranlators).
```
[the remark this answers is not in the document]
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:2113]: Resolved with #207.

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 23]
```diff
@@ -1927,82 +2054,94 @@ A CSAF content management system satisfies the "CSAF content management system"
 
 * When updating an exsting CSAF document:
```
[the remark this answers is not in the document]
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:2055]: Resolved with #207.

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 24]
```diff
@@ -137,14 +125,28 @@ For purposes of this document, the following terms and definitions apply:
...
 
 **CSAF post-processor**: CSAF producer that transforms an existing CSAF document into a new CSAF document, for example, by removing or redacting elements according to sharing policies.
 
 **CSAF producer**: program that emits output in the CSAF format
 
+**CSAF translator**: CSAF post-processor which takes a CSAF document as input and translates values of properties into another language. The output is a valid CSAF document.
+
+**CSAF viewer**: CSAF consumer that reads a CSAF document, displays a list of the results it contains, and allows an end user to view each result in the context of the artifact in which it occurs.
+
+**CVRF CSAF converter**: CSAF producer which takes a CVRF document as input and converts it into a vaild CSAF document.
```
[the remark this answers is not in the document]
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:148]: Resolved with #207.

File: csaf_2.0/json_schema/csaf_json_schema.json [hunk 25]
```diff
@@ -232,36 +232,36 @@
               "minLength": 4
             },
             "serial_numbers": {
-              "$comment": "TODO: Add remark in prose that this should only be used if a certain range of serial numbers with a software version is affected or the serial numbers change during update.",
+              "$comment": "TODO: Add remark in prose that this should only be used if a certain range of serial numbers with a software version is affected, or the serial numbers change during update.",
               "title": "List of serial numbers",
-              "description": "Contains a list of parts or a full serial numbers.",
+              "description": "Contains a list of parts, or a full serial numbers.",
```
[the remark this answers is not in the document]
Comment (tschmidtb51) [reply csaf_2.0/json_schema/csaf_json_schema.json:237]: Resolved with #207.

**Pull request merged**